### ISSUE TITLE
Error when using tip_priors matrix and check_input is TRUE

### ISSUE THREAD

**Issue by sdgamboa** [sdgamboa/castor#1]
Error seems to be due to this line: https://github.com/sdgamboa/castor/blob/ca6fa7a17c816413396829d3dd1ce400d000f82c/R/asr_mk_model.R#L57C22-L57C22

output of `&& (rownames(tip_priors)!=tree$tip.label))` is a vecotr of length > 1. 
It should be of length == 1 when using `&&`

``` r
library(ape)
library(castor)
#> Loading required package: Rcpp
set.seed(382)
tree <- rtree(10, rooted = TRUE)
tree$tip.label
#>  [1] "t2"  "t8"  "t7"  "t3"  "t5"  "t10" "t4"  "t1"  "t9"  "t6"
tip_priors <- matrix(
    data = rep(NA, Ntip(tree) * 4),
    ncol = 4, 
    dimnames = list(rownames = tree$tip.label, colnames = LETTERS[1:4])
)
tip_priors['t3', ] <- c(1, 0, 0, 0)
tip_priors['t5', ] <- c(0, 1, 0, 0)
tip_priors['t9', ] <- c(0, 0, 1, 0)
tip_priors['t2', ] <- c(0, 0, 0, 1)
result <- hsp_mk_model(
    tree = tree, Nstates = ncol(tip_priors), tip_priors = tip_priors, 
    tip_states = NULL,
    check_input = TRUE
)
#> Error in (!is.null(rownames(tip_priors))) && (!is.null(tree$tip.label)) && : 'length = 4' in coercion to 'logical(1)'
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.1 (2023-06-16)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2023-12-26
#>  pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  ape         * 5.7-1   2023-03-13 [1] RSPM (R 4.3.1)
#>  castor      * 1.7.11  2023-08-17 [1] RSPM (R 4.3.0)
#>  cli           3.6.2   2023-12-11 [1] RSPM (R 4.3.0)
#>  digest        0.6.33  2023-07-07 [1] CRAN (R 4.3.1)
#>  evaluate      0.23    2023-11-01 [1] RSPM (R 4.3.1)
#>  fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.3.1)
#>  fs            1.6.3   2023-07-20 [1] CRAN (R 4.3.1)
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.3.1)
#>  htmltools     0.5.7   2023-11-03 [1] RSPM (R 4.3.0)
#>  knitr         1.45    2023-10-30 [1] RSPM (R 4.3.1)
#>  lattice       0.22-5  2023-10-24 [2] RSPM (R 4.3.1)
#>  lifecycle     1.0.4   2023-11-07 [1] RSPM (R 4.3.0)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.3.1)
#>  Matrix        1.6-4   2023-11-30 [2] RSPM (R 4.3.1)
#>  naturalsort   0.1.3   2016-08-30 [1] RSPM (R 4.3.0)
#>  nlme          3.1-164 2023-11-27 [2] RSPM (R 4.3.1)
#>  purrr         1.0.2   2023-08-10 [1] RSPM (R 4.3.0)
#>  R.cache       0.16.0  2022-07-21 [1] RSPM (R 4.3.0)
#>  R.methodsS3   1.8.2   2022-06-13 [1] RSPM (R 4.3.0)
#>  R.oo          1.25.0  2022-06-12 [1] RSPM (R 4.3.0)
#>  R.utils       2.12.3  2023-11-18 [1] RSPM (R 4.3.0)
#>  Rcpp        * 1.0.11  2023-07-06 [1] CRAN (R 4.3.1)
#>  reprex        2.0.2   2022-08-17 [1] CRAN (R 4.3.1)
#>  rlang         1.1.2   2023-11-04 [1] RSPM (R 4.3.0)
#>  rmarkdown     2.25    2023-09-18 [1] RSPM (R 4.3.0)
#>  RSpectra      0.16-1  2022-04-24 [1] RSPM (R 4.3.0)
#>  rstudioapi    0.15.0  2023-07-07 [1] CRAN (R 4.3.1)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.3.1)
#>  styler        1.10.2  2023-08-29 [1] RSPM (R 4.3.0)
#>  vctrs         0.6.5   2023-12-01 [1] RSPM (R 4.3.1)
#>  withr         2.5.2   2023-10-30 [1] RSPM (R 4.3.1)
#>  xfun          0.41    2023-11-01 [1] RSPM (R 4.3.1)
#>  yaml          2.3.8   2023-12-11 [1] RSPM (R 4.3.0)
#> 
#>  [1] /home/user/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /home/user/apps/R-4.3.1/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

<sup>Created on 2023-12-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by sdgamboa** [comment]
Solved with PR #2 
No more errors when checking names:

``` r
library(ape)
library(castor)
#> Loading required package: Rcpp
set.seed(382)
tree <- rtree(10, rooted = TRUE)
tree$tip.label
#>  [1] "t2"  "t8"  "t7"  "t3"  "t5"  "t10" "t4"  "t1"  "t9"  "t6"
tip_priors <- matrix(
    data = rep(NA, Ntip(tree) * 4),
    ncol = 4, 
    dimnames = list(rownames = tree$tip.label, colnames = LETTERS[1:4])
)
tip_priors['t3', ] <- c(1, 0, 0, 0)
tip_priors['t5', ] <- c(0, 1, 0, 0)
tip_priors['t9', ] <- c(0, 0, 1, 0)
tip_priors['t2', ] <- c(0, 0, 0, 1)
result <- hsp_mk_model(
    tree = tree, Nstates = ncol(tip_priors), tip_priors = tip_priors, 
    tip_states = NULL,
    check_input = TRUE
)
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.1 (2023-06-16)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2023-12-26
#>  pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  ape         * 5.7-1   2023-03-13 [1] RSPM (R 4.3.1)
#>  castor      * 1.7.11  2023-08-17 [1] CRAN (R 4.3.1)
#>  cli           3.6.2   2023-12-11 [1] RSPM (R 4.3.0)
#>  digest        0.6.33  2023-07-07 [1] CRAN (R 4.3.1)
#>  evaluate      0.23    2023-11-01 [1] RSPM (R 4.3.1)
#>  fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.3.1)
#>  fs            1.6.3   2023-07-20 [1] CRAN (R 4.3.1)
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.3.1)
#>  htmltools     0.5.7   2023-11-03 [1] RSPM (R 4.3.0)
#>  knitr         1.45    2023-10-30 [1] RSPM (R 4.3.1)
#>  lattice       0.22-5  2023-10-24 [2] RSPM (R 4.3.1)
#>  lifecycle     1.0.4   2023-11-07 [1] RSPM (R 4.3.0)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.3.1)
#>  Matrix        1.6-4   2023-11-30 [2] RSPM (R 4.3.1)
#>  naturalsort   0.1.3   2016-08-30 [1] RSPM (R 4.3.0)
#>  nlme          3.1-164 2023-11-27 [2] RSPM (R 4.3.1)
#>  purrr         1.0.2   2023-08-10 [1] RSPM (R 4.3.0)
#>  R.cache       0.16.0  2022-07-21 [1] RSPM (R 4.3.0)
#>  R.methodsS3   1.8.2   2022-06-13 [1] RSPM (R 4.3.0)
#>  R.oo          1.25.0  2022-06-12 [1] RSPM (R 4.3.0)
#>  R.utils       2.12.3  2023-11-18 [1] RSPM (R 4.3.0)
#>  Rcpp        * 1.0.11  2023-07-06 [1] CRAN (R 4.3.1)
#>  reprex        2.0.2   2022-08-17 [1] CRAN (R 4.3.1)
#>  rlang         1.1.2   2023-11-04 [1] RSPM (R 4.3.0)
#>  rmarkdown     2.25    2023-09-18 [1] RSPM (R 4.3.0)
#>  RSpectra      0.16-1  2022-04-24 [1] RSPM (R 4.3.0)
#>  rstudioapi    0.15.0  2023-07-07 [1] CRAN (R 4.3.1)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.3.1)
#>  styler        1.10.2  2023-08-29 [1] RSPM (R 4.3.0)
#>  vctrs         0.6.5   2023-12-01 [1] RSPM (R 4.3.1)
#>  withr         2.5.2   2023-10-30 [1] RSPM (R 4.3.1)
#>  xfun          0.41    2023-11-01 [1] RSPM (R 4.3.1)
#>  yaml          2.3.8   2023-12-11 [1] RSPM (R 4.3.0)
#> 
#>  [1] /home/user/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /home/user/apps/R-4.3.1/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

<sup>Created on 2023-12-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by sdgamboa** [comment]
Check input still works.
``` r
library(ape)
library(castor)
#> Loading required package: Rcpp
set.seed(382)
tree <- rtree(10, rooted = TRUE)
tree$tip.label
#>  [1] "t2"  "t8"  "t7"  "t3"  "t5"  "t10" "t4"  "t1"  "t9"  "t6"
tip_priors <- matrix(
    data = rep(NA, Ntip(tree) * 4),
    ncol = 4, 
    dimnames = list(rownames = tree$tip.label, colnames = LETTERS[1:4])
)
tip_priors['t3', ] <- c(1, 0, 0, 0)
tip_priors['t5', ] <- c(0, 1, 0, 0)
tip_priors['t9', ] <- c(0, 0, 1, 0)
tip_priors['t2', ] <- c(0, 0, 0, 1)

rownames(tip_priors) <- sample(rownames(tip_priors), size = Ntip(tree), replace = FALSE)

result <- hsp_mk_model(
    tree = tree, Nstates = ncol(tip_priors), tip_priors = tip_priors, 
    tip_states = NULL,
    check_input = TRUE
)
#> Error in asr_mk_model(tree = known_subtree, tip_states = known_tip_states, : ERROR: Row names in tip_priors and tip labels in tree don't match
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.1 (2023-06-16)
#>  os       Pop!_OS 22.04 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/New_York
#>  date     2023-12-26
#>  pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  ape         * 5.7-1   2023-03-13 [1] RSPM (R 4.3.1)
#>  castor      * 1.7.11  2023-08-17 [1] CRAN (R 4.3.1)
#>  cli           3.6.2   2023-12-11 [1] RSPM (R 4.3.0)
#>  digest        0.6.33  2023-07-07 [1] CRAN (R 4.3.1)
#>  evaluate      0.23    2023-11-01 [1] RSPM (R 4.3.1)
#>  fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.3.1)
#>  fs            1.6.3   2023-07-20 [1] CRAN (R 4.3.1)
#>  glue          1.6.2   2022-02-24 [1] CRAN (R 4.3.1)
#>  htmltools     0.5.7   2023-11-03 [1] RSPM (R 4.3.0)
#>  knitr         1.45    2023-10-30 [1] RSPM (R 4.3.1)
#>  lattice       0.22-5  2023-10-24 [2] RSPM (R 4.3.1)
#>  lifecycle     1.0.4   2023-11-07 [1] RSPM (R 4.3.0)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.3.1)
#>  Matrix        1.6-4   2023-11-30 [2] RSPM (R 4.3.1)
#>  naturalsort   0.1.3   2016-08-30 [1] RSPM (R 4.3.0)
#>  nlme          3.1-164 2023-11-27 [2] RSPM (R 4.3.1)
#>  purrr         1.0.2   2023-08-10 [1] RSPM (R 4.3.0)
#>  R.cache       0.16.0  2022-07-21 [1] RSPM (R 4.3.0)
#>  R.methodsS3   1.8.2   2022-06-13 [1] RSPM (R 4.3.0)
#>  R.oo          1.25.0  2022-06-12 [1] RSPM (R 4.3.0)
#>  R.utils       2.12.3  2023-11-18 [1] RSPM (R 4.3.0)
#>  Rcpp        * 1.0.11  2023-07-06 [1] CRAN (R 4.3.1)
#>  reprex        2.0.2   2022-08-17 [1] CRAN (R 4.3.1)
#>  rlang         1.1.2   2023-11-04 [1] RSPM (R 4.3.0)
#>  rmarkdown     2.25    2023-09-18 [1] RSPM (R 4.3.0)
#>  RSpectra      0.16-1  2022-04-24 [1] RSPM (R 4.3.0)
#>  rstudioapi    0.15.0  2023-07-07 [1] CRAN (R 4.3.1)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.3.1)
#>  styler        1.10.2  2023-08-29 [1] RSPM (R 4.3.0)
#>  vctrs         0.6.5   2023-12-01 [1] RSPM (R 4.3.1)
#>  withr         2.5.2   2023-10-30 [1] RSPM (R 4.3.1)
#>  xfun          0.41    2023-11-01 [1] RSPM (R 4.3.1)
#>  yaml          2.3.8   2023-12-11 [1] RSPM (R 4.3.0)
#> 
#>  [1] /home/user/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /home/user/apps/R-4.3.1/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

<sup>Created on 2023-12-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
